ansible/collections/ansible_collections/nhsd/apigee/plugins/action/deploy_spec.py

```diff
@@ -47,7 +47,10 @@ def run(self, tmp=None, task_vars=None):
 
         existing_resources = [
             existing_resource
-            for existing_resource in task_vars["APIGEE_SPEC_RESOURCES"]
+            for existing_resource in (
+                task_vars.get("APIGEE_SPEC_RESOURCES")
+                or result["ansible_facts"]["APIGEE_SPEC_RESOURCES"]
```

Comment:
It's not immediately clear what this `or` is doing, and it might be more readable to rewrite this to

`task_vars.get("APIGEE_SPEC_RESOURCES", default=result["ansible_facts"]["APIGEE_SPEC_RESOURCES"])`